ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -167,6 +169,24 @@ public void execute(
           e.unauthorizedTopics()
       );
 
+      throw new KsqlException(createInsertFailedExceptionMessage(insertValues), rootCause);
+    } catch (final ClusterAuthorizationException e) {
+      // ClusterAuthorizationException is thrown when using idempotent producers
+      // and either a topic write permission or a cluster-level idempotent write
+      // permission (only applicable for broker versions no later than 2.8) is
+      // missing. In this case, we include additional context to help the user
+      // distinguish this type of failure from other permissions exceptions
+      // such as the ones thrown above when TopicAuthorizationException is caught.
+      final Exception rootCause = new KsqlTopicAuthorizationException(
+          AclOperation.WRITE,
+          Collections.singletonList(dataSource.getKafkaTopicName()),
+          // Ideally we would forward e.getMessage() instead of the hard-coded
+          // message below, but until this error message is improved on the Kafka
+          // side, e.getMessage() is not helpful. (Today it is just "Cluster
+          // authorization failed.")
+          "The producer is not authorized to do idempotent sends"
```

Comment:
Should we include how to disable idempotent? or should we include a server config that disables idempotent for producers so users don't need to do it every time they run INSERT VALUES?

Reply:
I'm not convinced it's necessary. The only cases in which a user would hit this is if one of the following is true:
1. They do not have write permissions to the topic.
2. They have write permissions, but they are using a broker version 2.8 or older and do not have IDEMPOTENT_WRITE permissions.
3. They are using a broker version 0.11 or older.

If it's the first, then disabling idempotent writes isn't going to help. The third should be extremely rare at this point, and we don't want to be in the business of supporting that. So that just leaves the second as a reason in which disabling the idempotent write would help. If you think it's helpful, we can add additional context here such as "The producer is not authorized to do idempotent sends. Check that you have write permissions to the specified topic, and disable idempotent sends by setting `enable.idempotent=false` if necessary."

Reply:
Updated the error message to the example in my comment above. Lmk what you think!